js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
         }
+
+
         if (!(layer instanceof StyleLayer)) {
+            if (validateStyle.emitErrors(this._style, validateStyle.layer({
```

Comment:
I agree that this is a better API design. Lets ticket it out and punt for the time being. The validator API probably won't see wide usage outside of GL JS. https://github.com/mapbox/mapbox-gl-style-spec/issues/416